### PR TITLE
JDK-8299396: Remove metaprogramming/removeExtent.hpp

### PR DESCRIPTION
**Pull request by jcking**
Code cleanup of pre-C++11 <type_traits> implementations.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299396](https://bugs.openjdk.org/browse/JDK-8299396): Remove metaprogramming/removeExtent.hpp

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11798/head:pull/11798` \
`$ git checkout pull/11798`

Update a local copy of the PR: \
`$ git checkout pull/11798` \
`$ git pull https://git.openjdk.org/jdk pull/11798/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11798`

View PR using the GUI difftool: \
`$ git pr show -t 11798`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11798.diff">https://git.openjdk.org/jdk/pull/11798.diff</a>

</details>

### CODE REVIEW

**Review by kimbarrett — APPROVED** [review]
Looks good.

File: src/hotspot/share/gc/z/zSafeDelete.hpp [hunk 1]
```diff
@@ -26,12 +26,13 @@
...
 #include "gc/z/zLock.hpp"
-#include "metaprogramming/removeExtent.hpp"
+
+#include <type_traits>
 
 template <typename T>
 class ZSafeDeleteImpl {
 private:
-  typedef typename RemoveExtent<T>::type ItemT;
+  typedef typename std::remove_extent<T>::type ItemT;
```
Comment (kimbarrett) [inline src/hotspot/share/gc/z/zSafeDelete.hpp:35]: `using ItemT = std::remove_extent_t<T>;`
e.g. use the `_t` suffixed trait, and prefer type aliases.

**Review by kimbarrett — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/z/zSafeDelete.hpp (hunk 1 above)
Comment (jcking) [reply src/hotspot/share/gc/z/zSafeDelete.hpp]: Done.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Review by walulyai — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]
/sponsor

**Pull request closed without merging**